File: 1677-matrix-diagonal-sum/1677-matrix-diagonal-sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        n = len(mat)
        result = 0
        for i in range(n):
            result += mat[i][i]
            result += mat[i][n-i-1]
        if n % 2 != 0:
            result -= mat[n//2][n//2]
        return result

File: 1677-matrix-diagonal-sum/1677-matrix-diagonal-sum.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int n = mat.length;
        int res = 0;
        for (int i = 0; i < n; i++) {
            res += mat[i][i] + mat[n-i-1][i];
        }
        if (n % 2 != 0) {
            res -= mat[n/2][n/2];
        }
        return res;
    }
}